Automattic/jetpack
# Dialogue Block: Add fallback for _experimentalNumberControl

Fixes #18285 

#### Changes proposed in this Pull Request:

* Abstracts the _experimentalNumberControl fallback used in extensions/shared/external-media/sources/google-photos/filter-option.js to extensions/shared/components/number-control.jsx
* Adds the fallback to the dialogue block
* Accounts for the possibility of `_experimentalNumberControl` becoming the publicly available API `NumberControl`

#### Jetpack product discussion


p4TIVU-9xW-p2

#### Does this pull request change what data or activity we track or use?


N/A

#### Testing instructions:






* Connect a test site to Jetpack and enable beta blocks.

*Dialogue Block*
* Add a Conversation Block to a post.
* Select one of the Dialogue Blocks in the Conversation Block and in the Block settings sidebar, check "Show conversation timestamps":
<img width="280" alt="Screen Shot 2021-01-11 at 5 36 27 PM" src="https://user-images.githubusercontent.com/1689238/104246271-b0ba5980-5433-11eb-9802-c192dbb48f03.png">

* Change the block timestamps in both the sidebar and in the block:

<img width="404" alt="Screen Shot 2021-01-11 at 5 36 47 PM" src="https://user-images.githubusercontent.com/1689238/104246333-cc256480-5433-11eb-899b-397428b8d13b.png">
<img width="278" alt="Screen Shot 2021-01-11 at 5 36 39 PM" src="https://user-images.githubusercontent.com/1689238/104246334-ccbdfb00-5433-11eb-8494-c210bcd06a09.png">

*Regression: Google Photos Filter*
* Add an Image Block to a post and click Select Image > Google Photos.
* Set date filter to "Specific Month/Year".
* Verify that the Month and Year inputs work correctly.
<img width="601" alt="Screen Shot 2021-01-11 at 5 42 38 PM" src="https://user-images.githubusercontent.com/1689238/104246662-69809880-5434-11eb-8c63-61305b6a0757.png">

#### Proposed changelog entry for your changes:


* Abstract fallback for unsafe _experimentalNumberControl API.
